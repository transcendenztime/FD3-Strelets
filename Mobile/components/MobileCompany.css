.MobileCompany
{
  /*border: double gray 3px;
  width: 680px;*/
  padding: 3px 7px;
  /*background-color: lightyellow;*/
}

.MobileCompanyClients
{
    display: inline-block;
    vertical-align: top;
}

.TableRight
{
    display: inline-block;
    vertical-align: top;
    padding: 10px;
    margin: 0 0 0 10px;
    border: double gray 6px;
    min-width: 475px;
}

.MobileCompanyClients table tr
{
  border-top: 1px solid gray;
}

.MobileCompanyClients table tr td
{
  padding: 10px 30px;
}

.MobileCompanyClients table
{
  border-collapse: collapse; 
}

.MobileCompanyName
{
  font-weight: bold;
}

.MobileCompanyClients
{
  padding-top: 20px;
}
