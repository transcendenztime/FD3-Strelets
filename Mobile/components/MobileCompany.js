import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        clientF: PropTypes.string.isRequired,
        clientI: PropTypes.string.isRequired,
        clientO: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    clients: this.props.clients,
  };

  setName1 = () => {
    this.setState({name:'МТС'});
  };

  setName2 = () => {
    this.setState({name:'Velcom'});
  };

  setBalance = (clientId,newBalance) => {
    let changed=false;
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      if ( c.id==clientId && c.balance!=newBalance ) {
        let newClient={...c}; // копия хэша изменившегося клиента
        newClient.balance=newBalance;
        newClients[i]=newClient;
        changed=true;
      }
    } );
    if ( changed )
      this.setState({clients:newClients});
  };
  
  
  setBalance1 = () => {
    this.setBalance(105,230);
  };

  setBalance2 = () => {
    this.setBalance(105,250);
  };

  render() {

    console.log("MobileCompany render");

    //console.log(this.props.clients);

    var clientsCode=this.state.clients.map( client =>
      <MobileClient key={client.id} info={client}  />
    );

    return (
      <div className='MobileCompany'>
        <input type="button" value="=МТС" onClick={this.setName1} />
        <input type="button" value="=Velcom" onClick={this.setName2} />
        <div className='MobileCompanyName'>Компания &laquo;{this.state.name}&raquo;</div>
        <div className='MobileCompanyClients'>
          {clientsCode}
        </div>
        <input type="button" value="Сидоров=230" onClick={this.setBalance1} />
        <input type="button" value="Сидоров=250" onClick={this.setBalance2} />
      </div>
    )
    ;

  }
  
}

export default MobileCompany;
/*import MobileClient from './MobileClient';

//////////////import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fio: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    clients: this.props.clients,
  };

  setName1 = () => {
    this.setState({name:'МТС'});
  };

  setName2 = () => {
    this.setState({name:'Velcom'});
  };
     
  setBalance = (clientId,newBalance) => {
    let changed=false;
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      if ( c.id==clientId && c.balance!=newBalance ) {
        let newClient={...c}; // копия хэша изменившегося клиента
        newClient.balance=newBalance;
        newClients[i]=newClient;
        changed=true;
      }
    } );
    if ( changed )
      this.setState({clients:newClients});
  };
  
  
  setBalance1 = () => {
    this.setBalance(105,230);
  };

  setBalance2 = () => {
    this.setBalance(105,250);
  };
  
  render() {

    console.log("MobileCompany render");

    var clientsCode=this.state.clients.map( client =>
      <MobileClient key={client.id} info={client}  />
    );

    return (
      <div className='MobileCompany'>
        <input type="button" value="=МТС" onClick={this.setName1} />
        <input type="button" value="=Velcom" onClick={this.setName2} />
        <div className='MobileCompanyName'>Компания &laquo;{this.state.name}&raquo;</div>
        <div className='MobileCompanyClients'>
          {clientsCode}
        </div>
        <input type="button" value="Сидоров=230" onClick={this.setBalance1} />
        <input type="button" value="Сидоров=250" onClick={this.setBalance2} />
      </div>
    )
    ;

  }

}

/////////////////export default MobileCompany;
*/