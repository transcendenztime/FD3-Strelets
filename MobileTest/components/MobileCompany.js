import React from 'react';
import PropTypes from 'prop-types';

import {mobileEvents} from './events';

import MobileClient from './MobileClient';
import ClientAddEdit from './ClientAddEdit';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    //name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        clientF: PropTypes.string.isRequired,
        clientI: PropTypes.string.isRequired,
        clientO: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    //name: this.props.name,
    clients: this.props.clients,
    clientToEdit: null,
    //mode: 0 - только таблица
    //mode: 1 - редактирование клиента
    //mode: 2 - добавление клиента
    mode: 0,
    clientsFilter: "all",
  };

  componentDidMount = () => {
    mobileEvents.addListener('EEditClient',this.editClient);
    mobileEvents.addListener('EDeleteRow',this.deleteRow);
    mobileEvents.addListener('ESave',this.save);
    mobileEvents.addListener('EAdd',this.add);
    mobileEvents.addListener('ECancel',this.cancel);
  };

  componentWillUnmount = () => {
    mobileEvents.removeListener('EEditClient',this.editClient);
    mobileEvents.removeListener('EDeleteRow',this.deleteRow);
    mobileEvents.removeListener('ESave',this.save);
    mobileEvents.removeListener('EAdd',this.add);
    mobileEvents.removeListener('ECancel',this.cancel);
  };

  clientsFilterAll = () => {
    this.setState({clientsFilter: "all"});
  }

  clientsFilterActive = () => {
    this.setState({clientsFilter: "active"});
  }

  clientsFilterBlocked = () => {
    this.setState({clientsFilter: "blocked"});
  }

  //сохраняем клиента
  save = (cliendData) => {
    let changed=false;

    let newClients=[...this.state.clients]; // копия самого массива клиентов
    let editIndex = newClients.findIndex(x => x.id === cliendData.id);
    //если изменилось хоть одно поле с данными клиента, тогда устанавливаем changed в "true",
    //чтобы затем записать новые данные в массив клиентов путем вызова setState()
    if(newClients[editIndex].id !== cliendData.id ||
      newClients[editIndex].clientF !== cliendData.clientF ||
      newClients[editIndex].clientI !== cliendData.clientI ||
      newClients[editIndex].clientO !== cliendData.clientO ||
      newClients[editIndex].balance !== cliendData.balance
    )
    {
      newClients[editIndex] = cliendData;
      changed = true;
    }
    console.log(cliendData);
    
    this.setState({clientToEdit: null,
      mode: 0,
    });

    if ( changed )
      this.setState({clients:newClients});
  }

  add = (cliendData) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.push(cliendData);
    this.setState({clientToEdit: null,
      mode: 0,
      clients:newClients,
    });
  }

  //отмена
  cancel = () => {
    this.setState({clientToEdit: null,
      mode: 0,
    });
  }

  editClient = (id) => {
    let clientToEditIndex = this.state.clients.findIndex(x => x.id === id);
    this.setState( {clientToEdit: clientToEditIndex,
      mode: 1,
    } );
  }

  addNewClient = () => {
    let clientToEditIndex = this.state.clients[this.state.clients.length - 1].id + 1;
    this.setState( {clientToEdit: clientToEditIndex,
      mode: 2,
    } );
  }

  deleteRow = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    //найдем индекс удаляемого клиента,
    //так как он не соответствует "id", который пришел из callback'а
    let deleteIndex = newClients.findIndex(x => x.id === id);
    newClients.splice(deleteIndex, 1);//удалим элемент (клиента) из массива
    this.setState( {clients:newClients} );
    //компонент перерисуется без карточки клиента
    this.setState( {mode:0} );
  }

  /*setName1 = () => {
    this.setState({name:'МТС'});
  };

  setName2 = () => {
    this.setState({name:'Velcom'});
  };*/

  render() {

    console.log("MobileCompany render");
    //console.log(this.props.clients);

    let clientsCode = [];
    
    this.state.clients.forEach( client => {
      if(this.state.clientsFilter == "all"){
        clientsCode.push(<MobileClient key={client.id} info={client}/>);
      }
      else if (this.state.clientsFilter == "active") {
        if(client.balance > 0){
          clientsCode.push(<MobileClient key={client.id} info={client}/>);
        }
      }
      else if(this.state.clientsFilter == "blocked"){
        if(client.balance <= 0) {
        clientsCode.push(<MobileClient key={client.id} info={client}/>);
        }
      }
    });

    return (
      <div className='MobileCompany'>
        {/*
        <input type="button" value="МТС" onClick={this.setName1} />
        <input type="button" value="Velcom" onClick={this.setName2} />
        <div className="MobileCompanyName">Компания &laquo;{this.state.name}&raquo;</div>
        */}
        <div className="FilterButtons">
          <button className="FilterAll" onClick={this.clientsFilterAll}>Все</button>
          <button className="FilterActive" onClick={this.clientsFilterActive}>Активные</button>
          <button className="FilterBlocked" onClick={this.clientsFilterBlocked}>Заблокированные</button>
        </div>
        <div className="MobileCompanyClients">
          <table>
            <thead>
              <tr className="ClientsHeader">
                  <td>Фамилия</td>
                  <td>Имя</td>
                  <td>Отчество</td>
                  <td>Баланс</td>
                  <td>Статус</td>
                  <td>Редактировать</td>
                  <td>Удалить</td>
              </tr>
            </thead>
            <tbody>
              {clientsCode}
            </tbody>
          </table>
          <input className="AddButton" type="button" value="Добавить клиента" onClick={this.addNewClient}/>
        </div>
        {
          (this.state.mode === 1 || this.state.mode === 2) &&
          <div className="TableRight">
            {
              (this.state.mode === 1)?
              <h2>Редактирование клиента</h2>
              :<h2>Добавление клиента</h2>
            }
            <ClientAddEdit key={this.state.clientToEdit}
              clients={this.state.clients}
              clientToEdit={this.state.clientToEdit}
              mode={this.state.mode}
            />
          </div>
        }
      </div> 
    );
  }
}

export default MobileCompany;