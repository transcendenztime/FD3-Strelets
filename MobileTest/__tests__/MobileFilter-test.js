"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';

test('MobileCompany нажали FilterAll', () => {

    let clientsArr=[ 
        {id:101, clientF:"Иванов", clientI:"Иван", clientO:"Иванович", balance:200}, 
        {id:105, clientF:"Сидоров", clientI:"Сидор", clientO:"Сидорович", balance:250}, 
        {id:110, clientF:"Петров", clientI:"Петр", clientO:"Петрович", balance:180},
        {id:120, clientF:"Григорьев", clientI:"Григорий", clientO:"Григорьевич", balance:-220},
      ];
  // создаём тестовую версию компонента
  const component = renderer.create(
    <MobileCompany
    clients={clientsArr}
    />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  //expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента саму кнопку "FilterAll"
  const buttonFilterAll = component.root.find( el => el.props.className=='FilterAll' /*&& el.props.aaa == 'bbb'*/ ); 
  // и "нажмём" на неё
  buttonFilterAll.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

});

test('MobileCompany нажали FilterActive', () => {

    let clientsArr=[ 
        {id:101, clientF:"Иванов", clientI:"Иван", clientO:"Иванович", balance:200}, 
        {id:105, clientF:"Сидоров", clientI:"Сидор", clientO:"Сидорович", balance:250}, 
        {id:110, clientF:"Петров", clientI:"Петр", clientO:"Петрович", balance:180},
        {id:120, clientF:"Григорьев", clientI:"Григорий", clientO:"Григорьевич", balance:-220},
      ];
  // создаём тестовую версию компонента
  const component = renderer.create(
    <MobileCompany
    clients={clientsArr}
    />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  //expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента саму кнопку "FilterActive"
  const buttonFilterActive = component.root.find( el => el.props.className=='FilterActive' /*&& el.props.aaa == 'bbb'*/ ); 
  // и "нажмём" на неё
  buttonFilterActive.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

});

test('MobileCompany нажали FilterBlocked', () => {

    let clientsArr=[ 
        {id:101, clientF:"Иванов", clientI:"Иван", clientO:"Иванович", balance:200}, 
        {id:105, clientF:"Сидоров", clientI:"Сидор", clientO:"Сидорович", balance:250}, 
        {id:110, clientF:"Петров", clientI:"Петр", clientO:"Петрович", balance:180},
        {id:120, clientF:"Григорьев", clientI:"Григорий", clientO:"Григорьевич", balance:-220},
      ];
  // создаём тестовую версию компонента
  const component = renderer.create(
    <MobileCompany
    clients={clientsArr}
    />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  //expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента саму кнопку "FilterBlocked"
  const buttonFilterBlocked = component.root.find( el => el.props.className=='FilterBlocked' /*&& el.props.aaa == 'bbb'*/ ); 
  // и "нажмём" на неё
  buttonFilterBlocked.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

});
