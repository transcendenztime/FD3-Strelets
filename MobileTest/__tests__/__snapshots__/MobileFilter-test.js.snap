// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileCompany нажали FilterActive 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="FilterButtons"
  >
    <button
      className="FilterAll"
      onClick={[Function]}
    >
      Все
    </button>
    <button
      className="FilterActive"
      onClick={[Function]}
    >
      Активные
    </button>
    <button
      className="FilterBlocked"
      onClick={[Function]}
    >
      Заблокированные
    </button>
  </div>
  <div
    className="MobileCompanyClients"
  >
    <table>
      <thead>
        <tr
          className="ClientsHeader"
        >
          <td>
            Фамилия
          </td>
          <td>
            Имя
          </td>
          <td>
            Отчество
          </td>
          <td>
            Баланс
          </td>
          <td>
            Статус
          </td>
          <td>
            Редактировать
          </td>
          <td>
            Удалить
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Иванов
          </td>
          <td
            className="Str"
          >
            Иван
          </td>
          <td
            className="Str"
          >
            Иванович
          </td>
          <td
            className="MobileClientBalance"
          >
            200
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceActive"
          >
            active
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Сидоров
          </td>
          <td
            className="Str"
          >
            Сидор
          </td>
          <td
            className="Str"
          >
            Сидорович
          </td>
          <td
            className="MobileClientBalance"
          >
            250
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceActive"
          >
            active
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Петров
          </td>
          <td
            className="Str"
          >
            Петр
          </td>
          <td
            className="Str"
          >
            Петрович
          </td>
          <td
            className="MobileClientBalance"
          >
            180
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceActive"
          >
            active
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
      </tbody>
    </table>
    <input
      className="AddButton"
      onClick={[Function]}
      type="button"
      value="Добавить клиента"
    />
  </div>
</div>
`;

exports[`MobileCompany нажали FilterAll 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="FilterButtons"
  >
    <button
      className="FilterAll"
      onClick={[Function]}
    >
      Все
    </button>
    <button
      className="FilterActive"
      onClick={[Function]}
    >
      Активные
    </button>
    <button
      className="FilterBlocked"
      onClick={[Function]}
    >
      Заблокированные
    </button>
  </div>
  <div
    className="MobileCompanyClients"
  >
    <table>
      <thead>
        <tr
          className="ClientsHeader"
        >
          <td>
            Фамилия
          </td>
          <td>
            Имя
          </td>
          <td>
            Отчество
          </td>
          <td>
            Баланс
          </td>
          <td>
            Статус
          </td>
          <td>
            Редактировать
          </td>
          <td>
            Удалить
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Иванов
          </td>
          <td
            className="Str"
          >
            Иван
          </td>
          <td
            className="Str"
          >
            Иванович
          </td>
          <td
            className="MobileClientBalance"
          >
            200
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceActive"
          >
            active
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Сидоров
          </td>
          <td
            className="Str"
          >
            Сидор
          </td>
          <td
            className="Str"
          >
            Сидорович
          </td>
          <td
            className="MobileClientBalance"
          >
            250
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceActive"
          >
            active
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Петров
          </td>
          <td
            className="Str"
          >
            Петр
          </td>
          <td
            className="Str"
          >
            Петрович
          </td>
          <td
            className="MobileClientBalance"
          >
            180
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceActive"
          >
            active
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Григорьев
          </td>
          <td
            className="Str"
          >
            Григорий
          </td>
          <td
            className="Str"
          >
            Григорьевич
          </td>
          <td
            className="MobileClientBalance"
          >
            -220
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceBlocked"
          >
            blocked
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
      </tbody>
    </table>
    <input
      className="AddButton"
      onClick={[Function]}
      type="button"
      value="Добавить клиента"
    />
  </div>
</div>
`;

exports[`MobileCompany нажали FilterBlocked 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="FilterButtons"
  >
    <button
      className="FilterAll"
      onClick={[Function]}
    >
      Все
    </button>
    <button
      className="FilterActive"
      onClick={[Function]}
    >
      Активные
    </button>
    <button
      className="FilterBlocked"
      onClick={[Function]}
    >
      Заблокированные
    </button>
  </div>
  <div
    className="MobileCompanyClients"
  >
    <table>
      <thead>
        <tr
          className="ClientsHeader"
        >
          <td>
            Фамилия
          </td>
          <td>
            Имя
          </td>
          <td>
            Отчество
          </td>
          <td>
            Баланс
          </td>
          <td>
            Статус
          </td>
          <td>
            Редактировать
          </td>
          <td>
            Удалить
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          className="MobileClient"
        >
          <td
            className="Str"
          >
            Григорьев
          </td>
          <td
            className="Str"
          >
            Григорий
          </td>
          <td
            className="Str"
          >
            Григорьевич
          </td>
          <td
            className="MobileClientBalance"
          >
            -220
          </td>
          <td
            className="MobileClientBalance MobileClientBalanceBlocked"
          >
            blocked
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td
            className="Control"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
      </tbody>
    </table>
    <input
      className="AddButton"
      onClick={[Function]}
      type="button"
      value="Добавить клиента"
    />
  </div>
</div>
`;
